### PR TITLE
fix(translate-json-object): make valid audit report

### LINKED ISSUE #10: Security report
Hi,

If i run `npm audit`i get a series of security issues connected with your dependencies, would you mind update your dependencies?

```
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > google-translate > request > hawk >  │
│               │ boom > hoek                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > google-translate > request > hawk >  │
│               │ cryptiles > boom > hoek                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > google-translate > request > hawk >  │
│               │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > google-translate > request > hawk >  │
│               │ sntp > hoek                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request > hawk >  │
│               │ boom > hoek                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request > hawk >  │
│               │ cryptiles > boom > hoek                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request > hawk >  │
│               │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ hoek                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request > hawk >  │
│               │ sntp > hoek                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/566                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ ReDoS via long string of semicolons                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tough-cookie                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.3.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request >         │
│               │ tough-cookie                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/130                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tough-cookie                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.3.3                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request >         │
│               │ tough-cookie                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/525                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Memory Exposure                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.6.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > google-translate > request >         │
│               │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/598                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Memory Exposure                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.6.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ translate-json-object                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ translate-json-object > yandex-translate > request >         │
│               │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/598                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

Note that the tunnel agent havnt fix their issues yet so you cant fix that though :)

### PR DESCRIPTION
**Pull request by sp90**
Fixes #10

### CODE REVIEW

**Comment by KhaledMohamedP** [comment]
Can you please resolve the travis-ci build issue, so I can merge this PR. Let me know if any action is required from my end. Thank you!

**Comment by sp90** [comment]
It can't be done some of the dev dependencies require higher than node 4.x - so it's time to leave this legacy version behind and focusing on newer versions

This is due to commitizen discarding support on node 4.x so i can't be done

File: package.json [hunk 1]
```diff
@@ -13,18 +13,18 @@
...
-    "commitizen": "2.9.3",
+    "commitizen": "^3.0.7",
     "cz-conventional-changelog": "1.2.0",
     "husky": "0.12.0",
-    "jest": "18.0.0",
-    "jsdoc-to-markdown": "2.0.1",
-    "semantic-release": "6.3.2",
+    "jest": "^24.4.0",
+    "jsdoc-to-markdown": "^4.0.1",
+    "semantic-release": "^15.13.4",
```
Comment (KhaledMohamedP) [inline package.json]: Can you please remove `^` from the dependencies versions

File: .travis.yml [hunk 2]
```diff
@@ -5,8 +5,9 @@ cache:
 notifications:
   email: false
 node_js:
+  - '10'
+  - '8'
   - '6'
-  - '4'
```
Comment (KhaledMohamedP) [inline .travis.yml:9]: Since this is breaking change, we should indicate this in the commit message

**Review by KhaledMohamedP — CHANGES_REQUESTED** [review]
Can you please address the comments below before I can merge this PR!

**Comment by KhaledMohamedP** [comment]
Since we are supporting newer version of `npm`, we need to commit the generated `package-lock.json` file too. 

I really appreciate your help here 😄

**Comment by sp90** [comment]
@KhaledMohamedP Added in the latest commit

**Comment by KhaledMohamedP** [comment]
@sp90 can you address the following comments

- https://github.com/KhaledMohamedP/translate-json-object/pull/16#discussion_r265406071
- https://github.com/KhaledMohamedP/translate-json-object/pull/16#discussion_r265405504

 also please fix Travis-CI, so I can merge in the code

**Comment by sp90** [comment]
@KhaledMohamedP Can't fix it for you because its the travis cache that dosn't match the package versions so you need to clear the travis caches or give me access

**Comment by KhaledMohamedP** [comment]
@sp90 I cleared the cache and re-build this PR and it worked. 

However `semantic-release` changed their API!

 I tested this command for publishing the package and it doesn't seems to work  anymore 

`"semantic-release": "semantic-release pre && npm publish && semantic-release post",`

We will need to accommodate for the new api changes if we want to update the `semantic-release` package 😞 

https://semantic-release.gitbook.io/semantic-release/usage/configuration

**Comment by sp90** [comment]
Please check the updates

**Comment by KhaledMohamedP** [comment]
Since the build is failing, I went ahead and created a PR that upgrades Node.js and some other core dependencies!

Thank you for addressing this issue 🙏

**Comment by sp90** [comment]
Thanks for taking away my contribution 👍 i just love to give away my time without getting credit

/Cheers

**Pull request closed without merging**